pkg/cstorpool/v1alpha2/kubernetes.go

```diff
@@ -88,24 +153,71 @@ func KubeClient(opts ...kubeclientBuildOption) *kubeclient {
 
 // getClientOrCached returns either a new instance
 // of kubernetes client or its cached copy
-func (k *kubeclient) getClientOrCached() (*clientset.Clientset, error) {
+func (k *Kubeclient) getClientOrCached() (*clientset.Clientset, error) {
 	if k.clientset != nil {
 		return k.clientset, nil
 	}
 	c, err := k.getClientset()
 	if err != nil {
-		return nil, err
+		return nil,
+			errors.WithStack(errors.Wrapf(err, "failed to get clientset:"))
 	}
 	k.clientset = c
 	return k.clientset, nil
 }
 
-// List returns a list of cstor pool
+// List returns a list of CStorPool
 // instances present in kubernetes cluster
-func (k *kubeclient) List(opts metav1.ListOptions) (*apis.CStorPoolList, error) {
-	cli, err := k.getClientOrCached()
+func (k *Kubeclient) List(opts metav1.ListOptions) (*apis.CStorPoolList, error) {
+	cs, err := k.getClientOrCached()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list CStorPool:")
+	}
+	return k.list(cs, opts)
+}
+
+// Get returns an CStorPool instance from kubernetes cluster
+func (k *Kubeclient) Get(name string, opts metav1.GetOptions) (*apis.CStorPool, error) {
+	if strings.TrimSpace(name) == "" {
+		return nil, errors.New("failed to get CStorPool: missing name")
+	}
+	cs, err := k.getClientOrCached()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get CStorPool: {%s}", name)
+	}
+	return k.get(cs, name, opts)
+}
+
+// Create creates an CStorPool instance in kubernetes cluster
+func (k *Kubeclient) Create(obj *apis.CStorPool) (*apis.CStorPool, error) {
+	cs, err := k.getClientOrCached()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create CStorPool:")
```

Comment:
```go
errors.Wrapf(err, "failed to create cstor pool: failed to get clientset: %s", obj)
```

Reply:
Updated with - 
```go
errors.Wrapf(err, "failed to create cstor pool: %s", obj)
```
as `failed to get clientset` message is present in error.